Signed policy symbol.

jvanz opened the pull request:
Adds a new symbol in the policy card to show when it is signed by a trusted entity. Now, when a policy is signed a green "check mark" is show in the policy card.

Related to https://github.com/kubewarden/policy-hub/issues/173

![image](https://user-images.githubusercontent.com/1514798/164225111-5c3755be-db03-4950-8aa3-51b4bbaf99ad.png)

jvanz committed:
Signed policy symbol.

Adds a new symbol in the policy card to show when it is signed by a
trusted entity.

ncounter reviewed (commented):
Overall it LGTM.

I only have two items to point out:
1- I wouldn't mind if the custom color was implemented following the [official way of doing it](https://mui.com/material-ui/customization/palette/#adding-new-colors) but I can understand it is a bit over-engineering
2- While I like the `check` icon, I would like to consider the [workspace premium](https://fonts.google.com/icons?selected=Material%20Icons%20Outlined%3Aworkspace_premium%3A) as well, IIUC the purpose of the new property

Feel free to consider or drop my two suggestions though.
:+1:

viccuad reviewed (commented):
LGTM too. I agree with the point about coloring. For the icon, apart from the `workspace_premium`, I would like to propose [verified](https://fonts.google.com/icons?selected=Material+Icons+Outlined:verified:) (like the Twitter tick), or [verified_user](https://fonts.google.com/icons?selected=Material+Icons+Outlined:verified_user:) which has a shield, or just a [key](https://fonts.google.com/icons?selected=Material+Icons+Outlined:key:). But don't want to derail with yak shaving, I'm happy with any.

jordojordo commented:
LGTM - I agree with adding the color property as well, and I'm partial to the [verified_user](https://fonts.google.com/icons?selected=Material+Icons+Outlined:verified_user:). I think keeping with the Kubewarden Shield motif would be nice.

jordojordo reviewed (commented):
LGTM - I agree with adding the color property to keep it consistent. Also I'm partial to the [verified_user](https://fonts.google.com/icons?selected=Material+Icons+Outlined:verified_user:) icon, I think keeping with the Kubewarden shield motif would be a nice touch.

flavio commented:
I vote for `verified_user` too :)

jvanz committed:
Use VerifiedUser icon.

Updates the Policy.js script to use the VerifiedUser icon to show that a
policy is signed by a trusted entity.

jordojordo commented:
Following a discussion with @jvanz and after further investigation - I think adding a class for the badge in `hub.css` would make the most sense.
A lot of the classes in hub.css could be a part of a theme, which if we decided to refactor how we implement the styling it would be nice to use the `ThemeProvider` component. But since `hub.css` is already being used in this way, adding a ThemeProvider for this one section may over-complicate things.

jvanz committed:
Customize a style for the signed policy symbol.

Changed the Policy.js script to use a custom style to define the color
used by the signed policy symbol.

jvanz commented:
I cannot not use a custom theme. For some reason, even when I changed the theme only for the signed policy icon component, all the icons change colors. :man_shrugging: .  Thus, I've create a custom style only for this icon. I hope this is enough. If you still believe that the theme should be changed. Please, let me know. I probably need your help to understand what I was doing wrong.

jvanz commented:
This is the UI with this PR:

![image](https://user-images.githubusercontent.com/1514798/164764670-20cd283c-3fce-4675-86d7-7648a9d8e56b.png)

jordojordo reviewed (approved):
LGTM - I think the change to add the `signedPolicyStyle` is an improvement. If more additions to this component are necessary in the future it would be nice to include the scaffolding for using a `ThemeProvider` when the time comes.

jvanz commented:
@ncounter any further comments? Thanks!

ncounter reviewed (approved):
LGTM!! :tada: 

Forget about the `ThemeProvider` for now, it's out of scope I agree. It can be part of a refactoring but it is not required for this and now.

I like how it looks like from the screenshot. Thank you for adding it. :+1:

jvanz commented:
Thank you all! Merging now!
